### PR TITLE
SONAR-6695 Remove execution of decorators from the batch

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/Decorator.java [hunk 1]
```diff
@@ -24,9 +24,12 @@
 
 /**
  * @since 1.10
+ * @deprecated since 5.2 there's no more decorator on batch side
+ * TODO add link to no API
```
Comment (henryju) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/Decorator.java]: ?

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java [hunk 2]
```diff
@@ -23,6 +23,8 @@
 
 /**
  * @since 1.11
+ * @deprecated since 5.2 there's no more decorator on batch side
+ * TODO add link to no API
```
Comment (henryju) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java]: s/no/new ?

**Comment by henryju** [comment]
LGTM, go go go !

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/Decorator.java (hunk 1 above)
Comment (julienlancelot) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/Decorator.java]: We should add a link to the new (not no !) API, but this API is not yet available

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java (hunk 2 above)
Comment (julienlancelot) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java]: indeed !

Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java [hunk 3]
```diff
@@ -23,6 +23,8 @@
...
  */
 public interface Formula {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java]: MAJOR Add the missing @Deprecated annotation. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/Decorator.java [hunk 4]
```diff
@@ -24,9 +24,12 @@
...
  */
 @BatchSide
 @ExtensionPoint
+@Deprecated
 public interface Decorator extends CheckProject {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/Decorator.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/Decorator.java (hunk 1 above)
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/Decorator.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java [hunk 5]
```diff
@@ -231,7 +231,10 @@ public Integer getId() {
 
   /**
    * @return the metric formula
+   * @deprecated since 5.2 there's no more decorator on batch side
+   * TODO add link to no API
    */
+  @Deprecated
   public Formula getFormula() {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java [hunk 6]
```diff
@@ -231,7 +231,10 @@ public Integer getId() {
 
   /**
    * @return the metric formula
+   * @deprecated since 5.2 there's no more decorator on batch side
+   * TODO add link to no API
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java [hunk 7]
```diff
@@ -241,7 +244,10 @@ public Formula getFormula() {
    *
    * @param formula the formula
    * @return this
+   * @deprecated since 5.2 there's no more decorator on batch side
+   * TODO add link to no API
    */
+  @Deprecated
   public Metric<G> setFormula(Formula formula) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java [hunk 8]
```diff
@@ -241,7 +244,10 @@ public Formula getFormula() {
    *
    * @param formula the formula
    * @return this
+   * @deprecated since 5.2 there's no more decorator on batch side
+   * TODO add link to no API
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

**Comment by SonarTech** [comment]
SonarQube analysis reported 9 issues:
* MAJOR 1 major
* INFO 8 info

Watch the comments in this conversation to review them.

**Comment by julienlancelot** [comment]
Thanks for your feedback @henryju. I'll push on master when my other PR (https://github.com/SonarSource/sonarqube/pull/434) will be merged.

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java [hunk 9]
```diff
@@ -23,7 +23,10 @@
 
 /**
  * @since 1.11
+ * @deprecated since 5.2 there's no more decorator on batch side
+ * TODO add link to new API
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java [hunk 10]
```diff
@@ -23,7 +23,10 @@
...
  */
+@Deprecated
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java]: MAJOR Add the missing @Deprecated annotation. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/Decorator.java [hunk 11]
```diff
@@ -24,9 +24,12 @@
 
 /**
  * @since 1.10
+ * @deprecated since 5.2 there's no more decorator on batch side
+ * TODO add link to new API
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/Decorator.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java [hunk 12]
```diff
@@ -231,7 +231,10 @@ public Integer getId() {
 
   /**
    * @return the metric formula
+   * @deprecated since 5.2 there's no more decorator on batch side
+   * TODO add link to new API
    */
+  @Deprecated
   public Formula getFormula() {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java [hunk 13]
```diff
@@ -231,7 +231,10 @@ public Integer getId() {
 
   /**
    * @return the metric formula
+   * @deprecated since 5.2 there's no more decorator on batch side
+   * TODO add link to new API
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java [hunk 14]
```diff
@@ -241,7 +244,10 @@ public Formula getFormula() {
    *
    * @param formula the formula
    * @return this
+   * @deprecated since 5.2 there's no more decorator on batch side
+   * TODO add link to new API
    */
+  @Deprecated
   public Metric<G> setFormula(Formula formula) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java [hunk 15]
```diff
@@ -241,7 +244,10 @@ public Formula getFormula() {
    *
    * @param formula the formula
    * @return this
+   * @deprecated since 5.2 there's no more decorator on batch side
+   * TODO add link to new API
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

**Comment by SonarTech** [comment]
SonarQube analysis reported 9 issues:
* MAJOR 1 major
* INFO 8 info

Watch the comments in this conversation to review them.

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java [hunk 16]
```diff
@@ -23,7 +23,10 @@
...
 public interface Formula {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/Formula.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 8 issues:
* INFO 8 info

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 12 issues:
* MINOR 4 minor
* INFO 8 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this unused import 'org.sonar.api.measures.PersistenceMode'.](https://github.com/SonarSource/sonarqube/blob/432ab0da084ef673d7d6f2dd1d6243620faccc10/sonar-batch/src/main/java/org/sonar/batch/sensor/DefaultSensorStorage.java#L53) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MINOR [Remove this unused import 'org.sonar.api.measures.SumChildDistributionFormula'.](https://github.com/SonarSource/sonarqube/blob/432ab0da084ef673d7d6f2dd1d6243620faccc10/sonar-batch/src/main/java/org/sonar/batch/sensor/DefaultSensorStorage.java#L54) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MINOR [Remove this unused import 'org.sonar.api.resources.Scopes'.](https://github.com/SonarSource/sonarqube/blob/432ab0da084ef673d7d6f2dd1d6243620faccc10/sonar-batch/src/main/java/org/sonar/batch/sensor/DefaultSensorStorage.java#L58) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MINOR [Remove this unused import 'org.sonar.api.measures.Formula'.](https://github.com/SonarSource/sonarqube/blob/432ab0da084ef673d7d6f2dd1d6243620faccc10/sonar-batch/src/main/java/org/sonar/batch/sensor/DefaultSensorStorage.java#L51) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 8 issues:
* INFO 8 info

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 8 issues:
* INFO 8 info

Watch the comments in this conversation to review them.

**Pull request merged**